pkg/utils/test/framework.go

```diff
@@ -190,3 +193,43 @@ func (f Framework) WaitUntilKeplerCondition(name string, t v1alpha1.ConditionTyp
 		})
 	return &k
 }
+
+func (f Framework) GetResourceNames(kind string) ([]string, error) {
+	res, err := oc.Get().Resource(kind, "").OutputJsonpath("{.items[*].metadata.name}").Run()
+	if err != nil {
+		return []string{}, err
+	}
+	nodes := strings.Split(res, " ")
+	return nodes, nil
+}
+
+func (f Framework) AddResourceLabels(kind, name string, l map[string]string) error {
+	b := new(bytes.Buffer)
+	for label, value := range l {
+		fmt.Fprintf(b, "%s=%s ", label, value)
+	}
+	return f.AddResourceLabelsStr(kind, name, b.String())
+}
+
+func (f Framework) AddResourceLabelsStr(kind, name, l string) error {
+	_, err := oc.Literal().From("oc label %s %s %s", kind, name, l).Run()
+	return err
+}
+
+func (f Framework) RemoveResourceLabels(kind, name string, l []string) error {
+	b := new(bytes.Buffer)
+	for _, label := range l {
+		fmt.Fprintf(b, "%s- ", label)
+	}
```

Comment:
how about using `strings.Builder` ?

Reply:
any issues with this?

Reply:
I don't think there is an issue with usage in test per se ... but I would heed - https://github.com/golang/go/blob/master/src/bytes/buffer.go#L65